### PR TITLE
AutoTest Enhancement: Implement "Invoke" to invoke methods and "GetProperty" to get just one single property

### PR DESCRIPTION
**Pull request by manish**

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AppResult.cs [hunk 1]
```diff
@@ -109,6 +110,15 @@ public virtual List<AppResult> Children (bool recursive = true)
 			return children;
 		}
 
+		public ObjectProperties GetProperty (object o, string propertyName)
+		{
+			LoggingService.LogDebug ($"[AppResult.GetProperty] Trying to fetch property '{propertyName}' on '{o.GetType().FullName}'");
+			var propertiesObject = new ObjectProperties ();
+			var result = GetPropertyAppResultFromObject (o, propertyName);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AppResult.cs]: Just return the AppResult?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AutoTestClientSession.cs [hunk 2]
```diff
@@ -383,6 +376,12 @@ public void Flash (Func<AppQuery, AppQuery> query)
 			}
 		}
 
+		public ObjectProperties GetProperty (Func<AppQuery, AppQuery> query, string propertyName)
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AutoTestClientSession.cs]: AppResult, not ObjectProperties.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AppResult.cs (hunk 1 above)
Comment (manish) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AppResult.cs]: Yes, Initially I thought of a way to return multiple properties at once by taking in `params string[] propertyNames`, but I think one property at a time is more sensible

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest.Results/GtkWidgetResult.cs [hunk 3]
```diff
@@ -526,11 +526,21 @@ public override void Flash ()
 			});
 		}
 
+		public override AppResult GetProperty(string propertyName)
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest.Results/GtkWidgetResult.cs]: These overrides are redundant

Comment (manish) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest.Results/GtkWidgetResult.cs]: The base method is different (different signature)

**Pull request closed without merging**